### ISSUE TITLE
Multiple documentation for some functions, some are even different

### ISSUE THREAD

**Issue by rohieb** [libopencm3/libopencm3#300]
I tried to implement SPI on STM31L1, and I noticed that the documentation for [`spi_init_master`](http://libopencm3.github.io/docs/latest/stm32l1/html/group__spi__file.html#ga40d1a1d70fd211d322d3957a792368ee) is given two times, and both describe different arguments of the function. Likewise, the [intro to SPI on STM32L1](http://libopencm3.github.io/docs/latest/stm32l1/html/group__spi__file.html#details) is given three times.

Relevant lines from IRC:
```
<karlp> yeah, that's because of how weird doxygen is with grouping things
<karlp> the L1 docs got a bit screwy as a lot of stuff was "group stm32f_spi"
    vs "group stm32l_spi" when they should have just been "group stm32_spi"
<karlp> and doxygen is all crazy.
```

**Comment by karlp** [comment]
I've started some work on this, and work will continue.  I'll close this when the work I have lands, but it's going to be an ongoing task unfortunately.

**Comment by karlp** [comment]
Fixed in all 3cc6ecd20330a2219f01a5b42ba5d6f36c232d0b
